### PR TITLE
Cu 8693z4817 awards

### PR DESCRIPTION
**Pull request by Woody1193**
This PR adds support for the `MarketQuery_AwardResultsQuery` endpoint. It also addresses a bug with deserialization that was resulting in validation messages being ignored on wrapped types and optional lists. Finally, this PR removes the `market_type` field from the `MarketQuery` DTO and makes it required on the `MarketCancel` DTO to align with the XML schema required by the MMS.

### CODE REVIEW

**Comment by Woody1193** [comment]
Task linked: [CU-8693z4817 MMS API service](https://app.clickup.com/t/8693z4817)

File: src/mms_client/types/enums.py [hunk 1]
```diff
@@ -19,8 +19,55 @@ class AreaCode(Enum):
     OKINAWA = "10"
 
 
+class Direction(Enum):
+    """Represents the reserve direction of the offer."""
+
+    SELL = "1"  # Increasing the reserves (sell)
+    # Note: Support for the BUY direction was removed from the MMS API
```
Comment (Liam-Woods) [inline src/mms_client/types/enums.py:26]: How are we supposed to "Buy low, sell high" if we aren't allowed to buy? smh my head.

File: src/mms_client/types/offer.py [hunk 2]
```diff
@@ -94,7 +66,7 @@ class OfferStack(Payload):
     tertiary_2_qty_kw: Optional[int] = power_positive("Tertiary2OfferQuantityInKw", True)
 
     # The unit price of the power, in JPY/kW/segment
-    unit_price: Decimal = price("OfferUnitPrice")
+    unit_price: Decimal = price("OfferUnitPrice", 10000.00)
```
Comment (Liam-Woods) [inline src/mms_client/types/offer.py:69]: Isn't the second argument of price the bool "optional"? What is going on here passing it 10,000?

File: src/mms_client/types/resource.py [hunk 3]
```diff
@@ -761,7 +733,7 @@ class ResourceData(Payload, tag="Resource"):
     has_contract: Optional[BooleanFlag] = attr(default=None, name="ContractExistence")
 
     # The maximum bid price for POWER_SUPPLY_1 power, in JPY/kW/hr
-    declared_maximum_unit_price_kWh: Annotated[Decimal, price("DeclaredMaximumUnitPrice", True)]
+    declared_maximum_unit_price_kWh: Annotated[Decimal, price("DeclaredMaximumUnitPrice", 10000.00, True)]
```
Comment (Liam-Woods) [inline src/mms_client/types/resource.py:736]: Samer here

File: src/mms_client/types/enums.py (hunk 1 above)
Comment (Woody1193) [reply src/mms_client/types/enums.py:26]: You sell hard enough that it overflows into a negative number and end up buying.

File: src/mms_client/types/offer.py (hunk 2 above)
Comment (Woody1193) [reply src/mms_client/types/offer.py:69]: No, the second argument is a floating-point value describing the upper limit of the field.

**Review by Liam-Woods — APPROVED** [review]

**Pull request merged**